### PR TITLE
Drop separate err and warnings on cache

### LINKED ISSUE #357: Merge `this.errs` and `this.warnings`
Make sure to include severity in the object.

### PR DESCRIPTION
**Pull request by SimenB**
Fixes #357 
Also a start for #356

It's mostly a subset of #274, so it's probably a bit rough around the edges

/cc @wojciechczerniak

### CODE REVIEW

**Comment by SimenB** [comment]
One of the tests seems to recurse infinitely, haven't been able to figure out why.

**Comment by SimenB** [comment]
There's a `this.cache.messages` that should be dropped, but I'll wait until I've refamiliarized myself with the code base

File: test/test.js [hunk 1]
```diff
@@ -252,54 +251,39 @@ describe( 'Core Methods: ', function() {
...
 
 		it( 'return correctly formatted msg', function() {
-			app.state.severity = 'Warning'
-			app.cache.file = 'testReporter'
-			app.cache.col = 0
-			app.cache.lineNo = 1
-			app.cache.origLine = 'Reporter Lyfe*'
 			app.cache.rule = 'universal'
-			var msg = 'universal disallowed'
 			var expectedOutput = '\u001b[4mtestReporter\u001b[24m\n1:0 \u001b[90muniversal\u001b[39m \u001b[33mwarning\u001b[39m universal disallowed'
```
Comment (SimenB) [inline test/test.js]: should just use `chalk.stripColor`, but whatever. Can fix when we refactor tests

Comment (wojciechczerniak) [inline test/test.js]: Yup, I expect tests to change a lot

**Comment by SimenB** [comment]
@wojciechczerniak Good point! `this.cache.messages` is already a thing, but I'll call it `messages` after I finish up #358 😄

**Pull request merged**